packages/server/lib/cloud/api.ts

```diff
@@ -452,23 +467,41 @@ module.exports = {
     responseCache = {}
   },
 
-  preflight (preflightInfo) {
+  postPreflight (preflightInfo) {
     return retryWithBackoff(async (attemptIndex) => {
-      const preflightBase = process.env.CYPRESS_API_URL ? apiUrl.replace('api', 'api-proxy') : apiUrl
-      const result = await rp.post({
-        url: `${preflightBase}preflight`,
-        body: {
-          apiUrl,
-          envUrl: process.env.CYPRESS_API_URL,
-          ...preflightInfo,
-        },
-        headers: {
-          'x-route-version': '1',
-          'x-cypress-request-attempt': attemptIndex,
-        },
-        json: true,
-        encrypt: 'always',
-      })
+      const { timeout, projectRoot } = preflightInfo
+
+      preflightInfo = _.omit(preflightInfo, 'timeout', 'projectRoot')
+
+      const preflightBaseProxy = apiUrl.replace('api', 'api-proxy')
+
+      const makeReq = async (baseUrl, agent) => {
+        return rp.post({
+          url: `${baseUrl}preflight`,
+          body: {
+            apiUrl,
+            ...await getEnvInformationForProjectRoot(projectRoot, process.pid.toString()),
+            ...preflightInfo,
+          },
+          headers: {
+            'x-route-version': '1',
+            'x-cypress-request-attempt': attemptIndex,
+          },
+          timeout: timeout ?? SIXTY_SECONDS,
+          json: true,
+          encrypt: 'always',
+          agent,
+        })
+      }
+
+      const postReqs = async () => {
+        return makeReq(preflightBaseProxy, null)
```

Comment:
very minor but can we pass an object literal instead of a primitive 2nd argument just to make this slightly clearer?

```js
return makeReq(preflightBaseProxy, { agent: null })
```

Reply:
Updated: [`e358a66` (#25837)](https://github.com/cypress-io/cypress/pull/25837/commits/e358a663e29e67a10a15cbd17f191d2f472d57b0)